### ISSUE TITLE
EnableCellular not enabled 

### ISSUE THREAD

**Issue by jacjac11** [leancodepl/patrol#1823]
### Steps to reproduce

 This bug arises only after version update.
 ```
 //Disconnect network
  await $.native.disableWifi();
  await $.native.disableCellular();

  Perform some actions here...

 // Connect network again
  await $.native.enableWifi();
  await $.native.enableCellular();
  
  Perform some actions that requires network here...
```

### Actual results

The network is not connected before executing the actions, thus failing.

Temporary added
` await $.pumpAndSettle(duration: const Duration(seconds: 10),timeout: Duration(seconds: 10));`
to work.

Expected: The await enableWifi / enableCellular should be connected first before return?

### Logs

<details>
<summary>Logs</summary>

```
<!-- Replace this line with your logs. Do not remove the backticks! -->
```

</details>

### Patrol version

patrol: ^2.3.1

### Patrol Doctor output

<details>
<summary>Patrol Doctor output</summary>

```
Patrol CLI version: 2.2.1
Program adb found in /usr/local/bin/adb
Env var $ANDROID_HOME is not set
Program xcodebuild found in /usr/bin/xcodebuild
Program ideviceinstaller not found (install with `brew install ideviceinstaller`)

```

</details>

### Flutter Doctor output

<details>
<summary>Flutter Doctor output</summary>

```
[✓] Flutter (Channel stable, 3.13.8, on macOS 13.0.1 22A400 darwin-x64, locale en-GB)
    • Flutter version 3.13.8 on channel stable 
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6c4930c4ac (4 days ago), 2023-10-18 10:57:55 -0500
    • Engine revision 767d8c75e8
    • Dart version 3.1.4
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /Users/jacquelinetan/Library/Android/sdk
    • Platform android-33, build-tools 33.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.79.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.68.0

[✓] Connected device (4 available)
    • SM N975F (mobile)                        • android-arm64  • Android 12 (API 31)
  

[✓] Network resources
    • All expected network resources are available.

```

</details>

**Comment by bartekpacia** [comment]
> This bug arises only after version update.

What was the previous version?

> Temporary added `await $.pumpAndSettle(duration: const Duration(seconds: 10),timeout: Duration(seconds: 10))` to work.

Maybe it takes a few seconds for the connection to become active? I think it's normal.

**Comment by jacjac11** [comment]
While testing on ios simulator I received this error, on android it works fine.
Could you advice?

flutter: Patrol (native): initialize() started
flutter: Patrol (native): initialize() succeeded
flutter: PatrolBinding: created
flutter: 00:00 +0: e2e_abc_test Abc test
flutter: Patrol (native): configure() started
flutter: Patrol (native): configure() succeeded
flutter: Patrol (native): enableWifi() started
flutter: Patrol (native): enableWifi() failed
flutter: ══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
flutter: The following PatrolActionException was thrown running a test:
flutter: Patrol action failed: NativeAutomatorClientException: enableWifi() failed with Invalid response: 400
flutter: The operation couldnât be completed. (patrol.PatrolError error 3.)

**Comment by bartekpacia** [comment]
Please share a reproducible example repo.

**Comment by bartekpacia** [comment]
Please provide a [minimal, reproducible sample](https://stackoverflow.com/help/minimal-reproducible-example).

Otherwise we cannot help you and the issue will be closed.

**Comment by bartekpacia** [comment]
Closing because of inactivity.

Feel free to open a new issue with all the required information provided, including a [minimal, reproducible sample](https://stackoverflow.com/help/minimal-reproducible-example).
